interfaces/Portalicious/src/app/pages/project/project-team/add-user-button/add-user-button.component.ts

```diff
@@ -1,23 +1,138 @@
+import {
+  injectMutation,
+  injectQuery,
+  injectQueryClient,
+} from '@tanstack/angular-query-experimental';
 import { ButtonModule } from 'primeng/button';
+import { DropdownModule } from 'primeng/dropdown';
+import { InputTextModule } from 'primeng/inputtext';
+import { MultiSelectModule } from 'primeng/multiselect';
+import { FormFieldWrapperComponent } from '~/components/form-field-wrapper/form-field-wrapper.component';
+import { FormSidebarComponent } from '~/components/form/form-sidebar.component';
+import { ProjectUserWithRolesLabel } from '~/models/project.model';
+import { ApiService } from '~/services/api.service';
 import { ToastService } from '~/services/toast.service';
+import {
+  generateFieldErrors,
+  genericFieldIsRequiredValidationMessage,
+} from '~/utils/form-validation';
+
+type AddUserToTeamFormGroup =
+  (typeof AddUserButtonComponent)['prototype']['formGroup'];
 
 @Component({
   selector: 'app-add-user-button',
+  styles: ``,
   standalone: true,
-  imports: [ButtonModule],
-  providers: [ToastService],
   templateUrl: './add-user-button.component.html',
-  styles: ``,
   changeDetection: ChangeDetectionStrategy.OnPush,
+  imports: [
+    ButtonModule,
+    FormSidebarComponent,
+    FormFieldWrapperComponent,
+    DropdownModule,
+    MultiSelectModule,
+    InputTextModule,
+    ReactiveFormsModule,
+  ],
 })
 export class AddUserButtonComponent {
+  projectId = input.required<number>();
+  projectUsers = input.required<ProjectUserWithRolesLabel[] | undefined>();
+  enableScope = input.required<boolean | undefined>();
+
+  private apiService = inject(ApiService);
   private toastService = inject(ToastService);
+  private queryClient = injectQueryClient();
+
+  formVisible = signal(false);
+
+  formGroup = new FormGroup({
+    userValue: new FormControl<number>(-1, {
+      // eslint-disable-next-line @typescript-eslint/unbound-method
+      validators: [Validators.required, Validators.min(0)],
+      nonNullable: true,
+    }),
+    rolesValue: new FormControl<string[]>([], {
+      // eslint-disable-next-line @typescript-eslint/unbound-method
+      validators: [Validators.required, Validators.minLength(1)],
+      nonNullable: true,
+    }),
+    scopeValue: new FormControl<string>('', {
+      validators: [Validators.pattern(/^\S*$/)],
+      nonNullable: true,
+    }),
+  });
+
+  formFieldErrors = generateFieldErrors<AddUserToTeamFormGroup>(
+    this.formGroup,
+    {
+      userValue: genericFieldIsRequiredValidationMessage,
+      rolesValue: genericFieldIsRequiredValidationMessage,
+      scopeValue: (control) => {
+        if (!control.invalid) {
+          return;
+        }
+        return $localize`Enter a valid scope. No spaces are allowed.`;
+      },
+    },
+  );
+
+  allUsers = injectQuery(this.apiService.getAllUsers());
+
+  unassignedUsers = computed(() => {
+    this.projectUsers();
+    const allUsers = this.allUsers.data();
+    if (!this.projectUsers() || !allUsers) {
```

Comment:
I think this would be a safer/better check perhaps? Not sure though

```suggestion
    if (this.allUsers.isPending() || !this.projectUsers.isPending()) {
```